unit_tests/src/test_json_apis_from_client.cpp

```diff
@@ -0,0 +1,110 @@
+#include "catch.hpp"
+
+#include "atomic_apply_metadata_operations.h"
+#include "connection_pool.hpp"
+#include "dstream.hpp"
+#include "filesystem.hpp"
+#include "irods_at_scope_exit.hpp"
+#include "irods_query.hpp"
+#include "replica.hpp"
+#include "rodsClient.h"
+#include "transport/default_transport.hpp"
+
+#include <boost/filesystem.hpp>
+#include <fmt/format.h>
+
+#include <chrono>
+#include <string_view>
+#include <string>
+#include <stdexcept>
+
+#include <unistd.h>
+#include <cstdlib>
+
+#include "json.hpp"
+
+
+namespace fs = irods::experimental::filesystem;
+
+using nlohmann::json;
+using namespace std::string_literals;
+
+namespace {
+
+    class error_setting_environment : public std::runtime_error {
+        public:
+        explicit error_setting_environment(const std::string & msg) : std::runtime_error{msg} {
+        }
+    };
+
+    void change_environment_variable (const char* env_var_name, char* & charptr_tmp, const char* new_value) {
+        charptr_tmp = getenv(env_var_name);
+        if (charptr_tmp) { charptr_tmp = strdup(charptr_tmp); }
+        if (0 != setenv(env_var_name,new_value,1)) {
+            throw error_setting_environment {"cannot change environment variable: "s + env_var_name};
+        }
+    }
+
+    void restore_environment_variable (const char *env_var_name, char* & charptr_tmp) {
+        if (charptr_tmp) {
+            if (0 != setenv(env_var_name, charptr_tmp, 1)) {
+                throw error_setting_environment {"cannot restore environment variable: "s + env_var_name};
+            }
+            free(charptr_tmp);
+            charptr_tmp = nullptr;
+        }
+    }
+}
+
+TEST_CASE("test atomic metadata with BinBytesBuf", "[xml]")
+{
+    char* save_protocol;
+
+    change_environment_variable("irodsProt",save_protocol,"1");
+
+    load_client_api_plugins();
+
+    auto conn_pool = irods::make_connection_pool(2);
```

Comment:
This creates two connections, but only one is used throughout the test.

Reply:
Should I leave this at a max 2 conns in the event the at_scope_exit brings up a new connection while deleting the sandbox, but before the other connection has been deleted?

Reply:
The code doesn't do anything asynchronously so one connection should be safe. You can reuse the connection handle obtained from the first call to `conn_pool->get_connection()`. You could replace the connection pool with `irods::experimental::client_connection` if you wanted. A connection pool isn't necessary for the test, but isn't wrong either.

Reply:
changed the pool to be size 1 and it hung ... not sure why .  (New issue possibly? ) But this unit test runs flawlessly and cleans up after itself as long as I keep the pool at size 2.